src/data/nat/relation.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2021 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import data.nat.basic
+import logic.relation
+/-!
+# Relations on `ℕ`
+
+This file contains properties about relations on `ℕ` and their closure operations
+(like the transitive closure).
+-/
+
+open set nat
+
+namespace relation
+
+/-- `(n, m)` is in the reflexive-transitive closure of `~` if `i ~ i+1` and `i+1 ~ i`
+  for all `i` between `n` and `m`. -/
+lemma refl_trans_gen_nat (r : ℕ → ℕ → Prop) {n m : ℕ}
```

Comment:
You should have `nat.refl_trans_gen_of_le` the one-sided version. Then this one is a one-liner. Same for `int`.